src/Sarif/Schemata/sarif-schema.json

```diff
@@ -2789,12 +2791,14 @@
 
         "nestingLevel": {
           "description": "An integer representing a containment hierarchy within the thread flow.",
-          "type": "integer"
+          "type": "integer",
+          "minimum": 0
         },
 
         "executionOrder": {
           "description": "An integer representing the temporal order in which execution reached this location.",
-          "type": "integer"
+          "type": "integer",
+          "minimum": 0
```

Comment:
>"minimum": 0 [](start = 10, length = 12)

how does this constraint help us? there's nothing in the spec that says these values are sequential. why should they be positive? what's the value?

Reply:
It's true, they don't have to be sequential. All the spec says is that they have to increase with time (that is, if two tFLs both have `executionOrder`, then the one that occurs at the later time must have the larger `executionOrder`.

---
In reply to: [276014542](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276014542) [](ancestors = 276014542)

Reply:
so what does the spec say? does it require this value to be positive? I don't see why it should.

---
In reply to: [276016264](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276016264) [](ancestors = 276016264,276014542)

Reply:
The spec actually does say they have to be positive (and if we believed that, the constraint would be "minimum: 1", not "minimum: 0"), but there's no reason for that constraint.

> A `threadFlowLocation` object **MAY** contain a property named `executionOrder` whose value is a positive integer that represents the temporal order in which execution reached this location, across all `threadFlowLocation` objects within all `threadFlow` objects belonging to a single codeFlow (§3.34).

Reply:
thanks, going to remove this.

---
In reply to: [276017630](https://github.com/Microsoft/sarif-sdk/pull/1428#discussion_r276017630) [](ancestors = 276017630)